### ISSUE TITLE
сюда постим поддомены, чтобы они попали в get_reestr_preresolved

### ISSUE THREAD

**Issue by bol-van** [bol-van/zapret#130]
Решил сделать свой сервис по обработке выгрузки реестра РКН.
Главная мотивация - отсутствие ipv6 в решениях от antifilter.network
Для получения листов используйте get_reestr_preresolved.sh и get_reestr_preresolved_smart.sh

На данный момент листы формируются из 3 источников :
1) Ресолвятся все домены из реестра блокированных сайтов РКН
2) Подход 1 не может учесть поддомены , не перечисленные в реестре. Их список приходится вести вручную.
Потому если вы вдруг встретились с таким случаем, что блокируемый поддомен не учтен в ipset, можете постить сюда название.   Поддомены легко выявить в броузере по кнопке F12.
3) Крупные сайты и хостинги используют прыгающие IP адреса. Что это такое ? Это когда вы заресолвили rutracker.org, и вам выдало одно. Через час - то же самое. А через день - другое. С другого провайдера или страны - другое.
В этом случае ipset будет работать нестабильно.
Я не знаю иного способа надежно решить вопрос, кроме как добавлять полностью диапазоны IP автономных систем.
На данный момент это cloudflare, twitter, facebook.
Если у вас есть что добавить к этому пункту - пишите.

get_reestr_preresolved.sh - это пункты 1 и 2
get_reestr_preresolved_smart.sh - это пункты 1, 2 и 3

**Comment by disappointed72** [comment]
Кстати заметил в дефолтной смарт выгрузке с антифильтра несколько префиксов гугла.. Похоже кто-то забавляется с заблоченными доменами. Случайно заметил в режиме дебага nfqws технические домены gmail. Пришлось выносить эти сети в exclude.

**Comment by bol-van** [comment]
Да, правда, это недостаток метода ресолва, с которым бороться достаточно сложно.
На этом в свое время погорел РКН, когда стали вносить IP адреса сбера, вконтакте и прочее.
Но в нашем случае самое плохое, что может случиться, это будет применено дурение к тому, к чему не должно.
Если это вызывает проблему - можно добавить в exclude

**Comment by chaojoy** [comment]
Спросите разработчика, как это программное обеспечение скрывает SNI и даже Client Hello веб-сайта, потому что скрытие TLS Client Hello очень полезно для обхода блокады провайдера. Другая проблема заключается в том, что использование этого программного обеспечения в маршрутизаторе OpenWrt вызовет резкое падение скорости сети.Какая конфигурация влияет на скорость сети?

**Comment by bol-van** [comment]
> Спросите разработчика, как это программное обеспечение скрывает SNI и даже Client Hello веб-сайта, потому что скрытие TLS Client Hello очень полезно для обхода блокады провайдера. Другая проблема заключается в том, что использование этого программного обеспечения в маршрутизаторе OpenWrt вызовет резкое падение скорости сети.Какая конфигурация влияет на скорость сети?

В отдельный топик pls. Здесь не место

**Comment by rockenren** [comment]
Для Twitter:
abs-0.twimg.com
abs-zero.twimg.com
abs.twimg.com
api.twitter.com
cdn.syndication.twimg.com
pbs-ak.twimg.com
pbs.twimg.com
platform.twitter.com
video-ak.twimg.com
video.twimg.com

Это как минимум.

Только с ними, в сочетании с get_reestr_preresolved_smart.sh, Twitter начал хотя бы открываться и частично показывать emdedded. И это без учета CDN с которых они видео показывают.

Я пытался ловить тупо через F12 в браузере в свое время, устал где-то на 100 записи в доменах, и это были только Facebook, Twitter, Instagram (все то, что показывает пустые квадраты вместо emdedded содержимого на большинстве сайтов).

Никаких нормальных вариантов получить тупо все поддомены для домена X я не нашел. Никаких crowdsourced списков тоже в итоге не нашел, все кто заявляли (например Censor Tracker) что добавили в свои списки "все диапазоны Twitter" в нормальном виде нигде эти списки не публиковали.

Я не знаю, может это только для меня проблема, но по факту опять приходится возвращаться (как было во времена когда реестр был небольшой) к policy-based заворачиванию через прокси/ss на уровне браузера или заворачивания всего трафика в vpn на уровне устройства (как во времена войны с Telegram)...

**Comment by bol-van** [comment]
> Это как минимум.

принято

> Я не знаю, может это только для меня проблема, но по факту опять приходится возвращаться (как было во времена когда реестр был небольшой) к policy-based заворачиванию через прокси/ss на уровне браузера или заворачиванию всего трафика в vpn на уровне устройства (как во времена войны с Telegram)...

можно использовать в zapret фильтрацию по хост листу вместо ip листа
там поддомены учитываются автоматом

**Comment by rockenren** [comment]
> можно использовать в zapret фильтрацию по хост листу вместо ip листа
там поддомены учитываются автоматом

да у меня даже с учетом обновленного роутера hostlist (и это только reestr_hostname_resolvable_ip4.txt) практически кладет его на лопатки по RAM.
OpenWrt 22.03.x, nftables и все его проблемы.

Спасибо за наводку, посмотрю как будет себя вести в режиме hostlist, при таких раскладах ipset просто теряет актуальность - самые "нужные" сайты в принципе сломаны из-за заблокированных технических поддоменов.

**Comment by bol-van** [comment]
> кладет его на лопатки по RAM

Есть 10-тысячник с antizapret. Он на nftables и 128 RAM нормально живет.
Если совсем плохо, можно отказаться от загружаемых листов и вручную набивать в юзер листы
Или по exclude работать, если что-то ломается с обходом

**Comment by bol-van** [comment]
Домен nethub.fi убился, поэтому листы переехали на гитхаб в репозиторий rulist.
Скрипты ipset обновлены

**Comment by y5vxv8kr4h-crypto** [comment]
не понимаю как запустить на mac помогите пожалуйста

**Comment by bol-van** [comment]
Источник выгрузки закрылся вместе с РосКомСвободой
